### PR TITLE
lint: Implement CSS linting with stylelint

### PR DESCRIPTION
**Pull request by vsvipul**
Adds linting to CSS files using stylelint.
Closes the CSS part of #676.

**What's this PR do?**
Implements CSS linting using [stylelint](https://github.com/stylelint/stylelint).
Format CSS file according to zulip webapp's [lint config](https://github.com/zulip/zulip/blob/master/.stylelintrc).

Some options have been excluded from lint config which are-
1. "color-hex-case": "lower"
2. "selector-pseudo-element-colon-notation": "double"
3. "color-no-hex": true
4. "color-named": "never"

These rules required changing more than just spaces and tabs, so i excluded them. I intend to add them separately later after this PR is merged.

**You have tested this PR on:**
  - [ ] Windows
  - [x] Linux/Ubuntu
  - [ ] macOS

### CODE REVIEW

File: package.json
Comment (kanishk98) [inline package.json]: I think it's important to Windows users that `xo` is run at the end. That test takes care of line endings, which can mess up quite often and I think they're taken care of by post-commit git hooks anyway. So if `xo` fails because of line endings, `stylelint` won't run and we don't want that.

File: package.json
Comment (kanishk98) [inline package.json]: These rules look good to me, just asking to confirm if there's been a discussion with the maintainers regarding them.

**Review by kanishk98 — APPROVED** [review]
Looks good to me. Fails and passes appropriately based on my testing. (Please do fix the Windows issue though).

File: package.json
Comment (vsvipul) [reply package.json]: I based the indentation rules on zulip webapp. I also intend to use the rules from the webapp once, these basic rules get merged.

File: package.json
Comment (vsvipul) [reply package.json]: hmm... If stylelint fails, would xo run ?

**Comment by vsvipul** [comment]
> Looks good to me. Fails and passes appropriately based on my testing. (Please do fix the Windows issue though).

@kanishk98 Thank u so much for your review. Which windows issue are you referring to ?

File: package.json
Comment (kanishk98) [reply package.json]: No, but then the code has actually failed the test. In the present case, even correctly styled code (which xo says is wrong but passes Travis tests) will not go through stylelint. So I think we should change the order.

**Comment by kanishk98** [comment]
> Which windows issue are you referring to ?

The one you've already addressed in the comments. :) [This one](https://github.com/zulip/zulip-electron/pull/699#discussion_r271947224).

File: package.json
Comment (vsvipul) [reply package.json]: ok got it.. will do.. thanks :)

**Comment by akashnimare** [comment]
Okay, boys - one suggestion. Since we have two linters now we could create a new script like -
```json
"lint": "stylelint app/renderer/css/*.css && xo",
"test: "lint"
```

**Comment by abhigyank** [comment]
Should we put the config in a `.stylelintrc` file or is it fine in the package.json?

**Comment by vsvipul** [comment]
@akashnimare i feel something like this will be better-
```
"lint-css": "stylelint app/renderer/css/*.css",
"lint-js": "xo",
"test: "lint-css && lint-js"
```
We can add `lint-html` later on .

**Comment by vsvipul** [comment]
@abhigyank we can use a separate file but there was no `.eslintrc` for xo, so i did not create one for stylelint as well.

**Comment by timabbott** [comment]
We should use a `.stylelintrc`.  Might be worth grabbing the config from the Zulip webapp so we use a consistent style.

**Comment by akashnimare** [comment]
@vsvipul can you create a `.eslintrc` for xo and .`stylelintrc` for stylelint?

**Comment by vsvipul** [comment]
> ```
> "lint-css": "stylelint app/renderer/css/*.css",
> "lint-js": "xo",
> "test: "lint-css && lint-js"
> ```
> We can add `lint-html` later on .

@akashnimare yeah. i'll work on adding those.. what do you think about the above package.json commands?

**Comment by akashnimare** [comment]
Looks good for now.

**Comment by vsvipul** [comment]
@akashnimare xo doesn't support a separate config file for now. It will add support for it later. Issue [here](https://github.com/xojs/xo/issues/330). I have created a separate .stylelintrc with similar config as zulip/zulip (remaining commands in top of this PR, will add them later, as they require some changes in code) and also changed the npm commands. Please review.

**Comment by vsvipul** [comment]
@akashnimare a reminder for this.

**Comment by akashnimare** [comment]
Merged. Thanks, everyone!

**Pull request merged**
